packages/good-design/src/apps/bridge/MicroBridge.tsx

```diff
@@ -0,0 +1,232 @@
+  text: string;
+  infoText?: string;
+  sourceChain: "fuse" | "celo";
+}) => (
+  <Stack mt="2" alignItems="center" direction={["column", "column", "row"]}>
+    <HStack alignItems="center" flex="2 0">
+      <Box>
+        <Status result={txStatus?.status} />
+      </Box>
+      <Box flex={"1 1"}>
+        <Text color="emerald.500" fontSize="md" ml="2">
+          {text}
+        </Text>
+      </Box>
+      {infoText && (
+        <Popover
+          trigger={triggerProps => {
+            return <IconButton {...triggerProps} icon={<InfoOutlineIcon />} />;
+          }}
+        >
+          <Popover.Content accessibilityLabel={infoText} w="md">
+            <Popover.CloseButton />
+            <Popover.Body>{infoText}</Popover.Body>
+          </Popover.Content>
+        </Popover>
+      )}
+    </HStack>
+    <Flex flex="1 1" alignItems="center" maxWidth="100%">
+      {txStatus && <ExplorerLink chainId={txStatus.chainId} addressOrTx={txStatus.transaction?.hash} />}
+    </Flex>
+  </Stack>
+);
+
+export const MicroBridge = ({
+  useBalanceHook,
+  useCanBridge,
+  onBridge,
+  onSetChain,
+  limits,
+  fees,
+  bridgeStatus,
+  relayStatus,
+  selfRelayStatus
+}: {
+  onBridge: OnBridge;
+  useBalanceHook: (chain: string) => string;
+  useCanBridge: (chain: string, amountWei: string) => { isValid: boolean; reason: string };
+  onSetChain?: (chain: string) => void;
+  limits?: {
+    [key: string]: { dailyLimit: BigNumber; txLimit: BigNumber; accountDailyLimit: BigNumber; minAmount: BigNumber };
+  };
+  fees?: { [key: string]: { minFee: BigNumber; maxFee: BigNumber; fee: BigNumber } };
+  bridgeStatus?: Partial<TransactionStatus>;
+  relayStatus?: Partial<TransactionStatus>;
+  selfRelayStatus?: Partial<TransactionStatus>;
+}) => {
+  const [isBridging, setBridging] = useState(false);
+  const [inputWei, setInput] = useState<string>("0");
+  const expectedFee = Number((Number(inputWei) * 0.001) / 100).toFixed(2);
+  const [enableTarget, setEnableTarget] = useState(false);
+
+  const [target, setTarget] = useState<string | undefined>();
+
+  const [sourceChain, setSourceChain] = useState<"fuse" | "celo">("fuse");
+
+  const targetChain = sourceChain === "fuse" ? "celo" : "fuse";
+  const balanceWei = useBalanceHook(sourceChain);
+
+  let { isValid, reason } = useCanBridge(sourceChain, inputWei);
+  const hasBalance = Number(inputWei) <= Number(balanceWei);
+  const isTargetValid = !enableTarget || isAddressValid(target || "");
+  const isValidInput = isValid && hasBalance && isTargetValid;
+  reason = reason || (!hasBalance && "balance") || (!isTargetValid && "target") || "";
+
+  const toggleChains = useCallback(() => {
+    setSourceChain(targetChain);
+    onSetChain && onSetChain(targetChain);
+  }, [setSourceChain, onSetChain, targetChain]);
+
+  const triggerBridge = useCallback(async () => {
+    setBridging(true);
+    try {
+      await onBridge(inputWei, sourceChain, enableTarget ? target : undefined);
+    } catch (e) {
```

Comment:
otherwise 'bridging' flag won't reset on success
also you don't need catch as you're doing nothing with the exception (e.g. logging etc)
```suggestion
    } finally {
```

Reply:
yeah i intentionally left it from finally. because otherwise it causes some blinking in setBridge.
so setbridging false is handled by the useEffect